app/src/main/java/de/westnordost/streetcomplete/quests/address/AddHousenumberForm.kt

```diff
@@ -49,13 +51,25 @@ class AddHousenumberForm : AbstractQuestFormAnswerFragment<HousenumberAnswer>()
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         val view = super.onCreateView(inflater, container, savedInstanceState)
         isHousename = savedInstanceState?.getBoolean(IS_HOUSENAME) ?: false
-        setLayout(if(isHousename) R.layout.quest_housename else R.layout.quest_housenumber)
+        isHousenameAndHousenumber = savedInstanceState?.getBoolean(IS_HOUSENAME_AND_HOUSENUMBER) ?: false
+        when {
+            isHousename -> {
+                setLayout(R.layout.quest_housename)
+            }
+            isHousenameAndHousenumber -> {
+                setLayout(R.layout.quest_housename_and_housenumber)
+            }
+            else -> {
+                setLayout(R.layout.quest_housenumber)
+            }
+        }
```

Comment:
On mobile so can't use the "suggest" feature, but I might pull out the `setlayout`

```kt
val layout = when {
  isHousename -> R.layout.quest_housename
  isHousenameAndHousenumber -> R.layout.quest_housename_and_housenumber
  else -> R.layout.quest_housenumber
}
setLayout(layout)
```

Reply:
That'd increase the readability a lot! But I think there is no need for the intermediate variable, so better:

```kt
setLayout(when {
  isHousename -> R.layout.quest_housename
  isHousenameAndHousenumber -> R.layout.quest_housename_and_housenumber
  else -> R.layout.quest_housenumber
})
```

Reply:
👍